File: VB/WindowsFormsApplication25/Program.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.Xpo

Namespace WindowsFormsApplication25

    Friend Module Program

        ''' <summary>
        ''' The main entry point for the application.
        ''' </summary>
        <STAThread>
        Sub Main()
            'XpoDefault.DataLayer = XpoDefault.GetDataLayer(DevExpress.Xpo.DB.MSSqlConnectionProvider.GetConnectionString("localhost", "PreviewPropertyChanges"), DevExpress.Xpo.DB.AutoCreateOption.DatabaseAndSchema);
            XpoDefault.DataLayer = New SimpleDataLayer(New DB.InMemoryDataStore())
            Call CreateSampleData()
            Call Application.EnableVisualStyles()
            Application.SetCompatibleTextRenderingDefault(False)
            Call Application.Run(New Form1())
        End Sub

        Private Sub CreateSampleData()
            Using uow As UnitOfWork = New UnitOfWork()
                If uow.FindObject(Of DomainObject)(Nothing) Is Nothing Then
                    Dim n As Integer = 13
                    For i As Integer = 0 To n - 1
                        Dim obj As DomainObject = New DomainObject(uow)
                        obj.IntProperty = i
                        obj.DecimalProperty = CDec(i) / n
                        obj.StringProperty = String.Format("sample{0}", i)
                        obj.DateTimeProperty = Date.Today.AddDays(i)
                    Next

                    uow.CommitChanges()
                End If
            End Using
        End Sub
    End Module
End Namespace

File: CS/WindowsFormsApplication25/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.Xpo;

namespace WindowsFormsApplication25 {
    static class Program {
        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main() {

            //XpoDefault.DataLayer = XpoDefault.GetDataLayer(DevExpress.Xpo.DB.MSSqlConnectionProvider.GetConnectionString("localhost", "PreviewPropertyChanges"), DevExpress.Xpo.DB.AutoCreateOption.DatabaseAndSchema);
            XpoDefault.DataLayer = new SimpleDataLayer(new DevExpress.Xpo.DB.InMemoryDataStore());
            CreateSampleData();

            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new Form1());
        }

        private static void CreateSampleData() {
            using (UnitOfWork uow = new UnitOfWork()) {
                if (uow.FindObject<DomainObject>(null) == null) {
                    int n = 13;
                    for (int i = 0; i < n; i++) {
                        DomainObject obj = new DomainObject(uow);
                        obj.IntProperty = i;
                        obj.DecimalProperty = (decimal)i / n;
                        obj.StringProperty = string.Format("sample{0}", i);
                        obj.DateTimeProperty = DateTime.Today.AddDays(i);
                    }
                    uow.CommitChanges();
                }
            }
        }
    }
}
